### PR TITLE
mkdocs-rss-plugin: init at 1.8.0

### PR DESCRIPTION
**Pull request by caarlos0**
## Description of changes

It's needed for mkdocs-material blog plugin.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by caarlos0** [comment]
can someone review this?

**Comment by caarlos0** [comment]
gently ping :)

**Comment by caarlos0** [comment]
gently ping @dansbandit

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+}:
+
+buildPythonPackage {
+  pname = "mkdocs-rss-plugin";
+  version = "1.8.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:8]:
```suggestion
  version = "1.8.0";
  pyproject = true;

  disabled = pythonOlder "3.8";
```

see https://github.com/Guts/mkdocs-rss-plugin/blob/main/setup.py#L82

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Guts";
+    repo = "mkdocs-rss-plugin";
+    rev = "1.8.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:13]:
```suggestion
    rev = "refs/tags/${version}";
```

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "sha256-rCz1Uk5uqIsnIWw0b1oBsjAO6aK/tpVgqAX/8dVnAGw=";
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:14]: SRI hashe should be using `hash`.

```suggestion
    hash = "sha256-rCz1Uk5uqIsnIWw0b1oBsjAO6aK/tpVgqAX/8dVnAGw=";
```

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:16]: [Run-time requirements](https://github.com/Guts/mkdocs-rss-plugin/blob/main/requirements/base.txt) should go into `propagatedBuildInputs`.

```suggestion
  nativeBuildInputs = [
    setuptools
  ];

  propagatedBuildInputs = [
    ...
  ];
```

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:18]: The upstream repo contains [tests](https://github.com/Guts/mkdocs-rss-plugin/tree/1.12.1/tests). `pytestCheckHook` could help here.

```text
  nativeCheckInputs = [
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  meta = with lib; {
+    description = "MkDocs plugin to generate a RSS feeds for created and updated pages, using git log and YAML frontmatter (page.meta).";
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:21]:
```suggestion
    description = "Plugin to generate a RSS feeds for created and updated pages";
    longDescription = ''
      MkDocs plugin to generate a RSS feeds for created and updated pages, using
      git log and YAML frontmatter (page.meta).
    '';
```

File: pkgs/development/python-modules/mkdocs-rss-plugin/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    homepage = "https://github.com/Guts/mkdocs-rss-plugin";
```
Comment (fabaff) [inline pkgs/development/python-modules/mkdocs-rss-plugin/default.nix:22]:
```suggestion
    homepage = "https://github.com/Guts/mkdocs-rss-plugin";
    changelog = "https://github.com/Guts/mkdocs-rss-plugin/blob/${version}/CHANGELOG.md";
```

**Review by fabaff — COMMENTED** [review]
Commit for Python modules should be prefixed with `python31XPackages.`.

**Comment by caarlos0** [comment]
ah, just noticed its already on master, closing, sorry for the delay


https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/mkdocs-rss-plugin/default.nix

**Pull request closed without merging**